### PR TITLE
Fix for Issue-19. Add Student death reporting support.

### PR DESCRIPTION
**Pull request by mmarum-sugarcrm**

### CODE REVIEW

File: package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+ // created: 2018-01-31 21:21:45
+$dictionary['Contact']['fields']['cause_of_death_c']['labelValue']='Cause of Death';
+$dictionary['Contact']['fields']['cause_of_death_c']['full_text_search']=array (
+  'enabled' => '0',
+  'boost' => '1',
+  'searchable' => false,
```
Comment (lschaefer-sugarcrm) [inline package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php]: I'm curious why this is not searchable?

File: package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+);
+$dictionary['Contact']['fields']['cause_of_death_c']['enforced']='';
+$dictionary['Contact']['fields']['cause_of_death_c']['dependency']='';
+
+ ?>
```
Comment (lschaefer-sugarcrm) [inline package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php]: :open_mouth:

File: package/src/custom/modules/Contacts/clients/base/views/list/list.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?php
```
Comment (lschaefer-sugarcrm) [inline package/src/custom/modules/Contacts/clients/base/views/list/list.php]: I'm curious what this file is doing?

File: docs/UseCases.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+## About Professor M's School for Gifted Coders
+Professor M aka Professor Marum has created an exclusive not-for-profit school for gifted coders. Coders are a subspecies of human who are born with superhuman capabilities in technology and analysis. 
+
+Professor M is gathering the best and brightest to train them to do battle against the forces of evil (specifically bad customer experiences and poor project implementations). It is hazardous and demanding work. 
+
+Professor M has deployed Sugar as a platform to help him manage both students and faculty so he can focus on teaching.
+
+##High Level Use Cases
```
Comment (lschaefer-sugarcrm) [inline docs/UseCases.md]: This is a really good idea.  Do we want to add high level implementation details here or link to the relevant PR/commits?  If I wanted to do implement something similar, I'd want to know what you did in Studio and how you knew how to implement the rest of the changes (related Sugar Doc or references).

Comment (lschaefer-sugarcrm) [reply docs/UseCases.md]: Also, we should reference this in the README

File: docs/UseCases.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+**Background**
+
+When doing battle against the forces of evil, it is possible that a coder is injured, comatose, or even (temporarily) killed. Professor M needs to report on the cause of death of any coder who has died in order to prevent the same horror from happening again.
+
+**Implementation Details**
+
+A custom dropdown field `vital_c` was added to the Students module and the Record view. The following vital statuses are selectable.
+    
+    'active' => 'Active'
```
Comment (lschaefer-sugarcrm) [inline docs/UseCases.md]: We may want to update the Postman collection to have this data

File: docs/UseCases.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    'injured' => 'Injured'
+    'comatose' => 'Comatose'
+    'deceased' => 'Deceased'
+
+When `deceased` is selected, a new Death Information panel appears on Record view and users are required to enter a `cause_of_death_c`. This field is reportable.
```
Comment (lschaefer-sugarcrm) [inline docs/UseCases.md]: I tested this out, and the panel appears and disappears well.  The only thing that surprised me is that if I mark a student as deceased, input their cause of death, save, update the student to be active, save, and update the student to be deceased, the original cause of death is displayed.  You could make a case for the original cause of death being stored or not.  My bigger concern is that if I were to mark a student as deceased, change back to active, and pull the data out of a report, search, or the db, is it possible a student could be active with a cause of death?

[the remark this answers is not in the document]
Comment (mmarum-sugarcrm) [reply docs/UseCases.md]: It's the default but it is something you can control in Studio. Same with auditability.

File: package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php (hunk 1 above)
Comment (mmarum-sugarcrm) [reply package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php]: I think custom fields are not searchable by default? This is generated by Studio.

File: package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php (hunk 2 above)
Comment (mmarum-sugarcrm) [reply package/src/custom/Extension/modules/Contacts/Ext/Vardefs/sugarfield_cause_of_death_c.php]: Generated file!

**Pull request merged**